app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperDownloader.kt

```diff
@@ -50,9 +51,17 @@ class WallpaperDownloader(
                     logger.error("Download response failure code: ${it.status}")
                     return@withContext
                 }
-                File(localFile.path.substringBeforeLast("/")).mkdirs()
-                it.body.useStream { input ->
-                    input.copyTo(localFile.outputStream())
+                try {
+                    File(localFile.path.substringBeforeLast("/")).mkdirs()
+                    it.body.useStream { input ->
+                        input.copyTo(localFile.outputStream())
+                    }
+                } catch (exception: IOException) {
+                    logger.error(
+                        exception.message ?: "Download failed: no throwable message included.",
+                        exception
+                    )
+                    crashReporter.submitCaughtException(exception)
```

Comment:
This could be a `map` on the `Result.runCatching` block, so that the `onFailure` block is shared. Then we aren't mixing our exception handling patterns.

Alternatively, the whole thing could be switching to a `try catch`

Reply:
Updated!